### DISCUSSION TITLE
Score file not getting generated even after successful run

### DISCUSSION THREAD

**Discussion by intomybioverse** [RosettaCommons/rosetta#76] [RosettaScripts] [answered]
I am trying to change the specificity of a protein so that it binds to a desired DNA sequence. For this, I am following the protocol mentioned below.

Thyme S, Baker D. Redesigning the specificity of protein-DNA interactions with Rosetta. Methods Mol Biol. 2014;1123:265-82. doi: 10.1007/978-1-62703-968-0_17. PMID: 24510272.

The program is sucessfully running, PDB files are getting generated but the score.sc file is not getting formed. 
I have referred to the guidelines on the basic rosetta scoring function (https://docs.rosettacommons.org/demos/latest/tutorials/scoring/scoring), used the commands

-out:file:scorefile output_files/score.sc
-out:path:scorefile /my_output_score/my_score.sc

However, score.sc file is still not created at the desired location. 

[ ERROR ]: Caught exception:

File: src/utility/options/OptionCollection.cc:1467
Option matching -out:path:scorefile not found in command line top-level context
Did you mean:
	 -chemically_conjugated_docking:scorefilter
	 -flexPepDocking:score_filter
	 -out:file:scorefile
	 -run:no_scorefile

AN INTERNAL ERROR HAS OCCURED. PLEASE SEE THE CONTENTS OF ROSETTA_CRASH.log FOR DETAILS.

Here is my rosetta_crash.log file for your reference.

Rosetta crash log. Please submit the contents of this file to http://crash.rosettacommons.org/
or use the script Rosetta/main/source/scripts/python/public/report_crashes.py

[START_CRASH_REPORT]
[ROSETTA_VERSION]: 2024.09+release.06b3cf8
[COMMIT_DATE]: 2024-03-07T14:59:22.624665
[APPLICATION]: UNKNOWN APPLICATION
[MODE]: Release
[EXTRAS]: default
[OS]: GNU/Linux
[COMPILER]: GCC version "11.4.0"
[STDLIB]: libstdc++ version 20230528
[START_OPTIONS]
NO OPTIONS SPECIFIED

[END_OPTIONS]

[START_BACKTRACE]: RAW_LIBC
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(backtrace_string[abi:cxx11](int)+0x5a) [0x7d3d615ab35a]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::excn::Exception::Exception(char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xe0) [0x7d3d615e9310]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(+0x19d54e) [0x7d3d6159d54e]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::options::OptionCollection::load_option_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::list<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool)+0x1e7) [0x7d3d616ee537]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::options::OptionCollection::load_options_from_stream(std::istream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)+0x126e) [0x7d3d616ef91e]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::options::OptionCollection::load_options_from_file_exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x1d8) [0x7d3d616efef8]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::options::OptionCollection::load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::list<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, bool)+0x221) [0x7d3d616f02e1]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::options::OptionCollection::load(int, char* const*, bool)+0x1db) [0x7d3d616f0c2b]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libcore.6.so(core::init::init_options(int, char**)+0x24) [0x7d3d5d9f26c4]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libcore.6.so(core::init::init(int, char**)+0x31) [0x7d3d5d9f6921]
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease(+0x34aa) [0x558dddecc4aa]
/lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7d3d60c29d90]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7d3d60c29e40]
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease(+0x35f5) [0x558dddecc5f5]

[END_BACKTRACE]

[FILE]: src/utility/options/OptionCollection.cc
[LINE]: 1467
[START_MESSAGE]
Option matching -out:path:scorefile not found in command line top-level context
Did you mean:
	 -chemically_conjugated_docking:scorefilter
	 -flexPepDocking:score_filter
	 -out:file:scorefile
	 -run:no_scorefile

[END_MESSAGE]
[END_CRASH_REPORT]

**Comment by roccomoretti** [comment, marked as answer]
First off, `-out:path` is a namespace that's used to control directories for output. But that's typically only for when you're outputting multiple files. If there's a single file associated with your output, it's usually the `-out:file` option (e.g. `-out:file:scorefile`) that is used. -- just omit the -out:path:scorefile

The other thing to keep in mind is that Rosetta will typically not create directories if it doesn't already exist. So if you have `-out:file:scorefile output_files/score.sc`, then the directory `output_files/` (relative to the running directory) must already exist. If you want to use `my_output_score/` as the directory instead, try `-out:file:scorefile my_output_score/my_score.sc`  -- note that `-out:file:scorefile /my_output_score/my_score.sc`  with the leading slash will interpret `/my_output_score` as an absolute path (rather than a relative path from the working directory), which will almost certainly fail, as no computer would have a `/my_output_score` directory at the root of the filesystem.

The protocol you listed uses the rosetta_scripts application, which should definitely output a scorefile at the path specified by `-out:file:scorefile` -- if you're still having trouble, posting the full command line you're using might help debug it.

**Reply by intomybioverse** [reply]
I tried this -out:file:scorefile my_output_score/my_score.sc, 
my_output_score is the name of a direcotory inside my working directory (the terminal was opened from here), still no score file named my_score.sc was generated. 
The program ran successfully but the .pdb files also got generated in the working directory. How to get the score.sc file separately and get the .pdb output files in a different directory?

**Reply by roccomoretti** [reply]
Rosetta generally won't create directories -- any directories need to already be existing. So you'd have to make the `my_output_score/` directory prior to starting Rosetta. If that directory exists, then the `-out:file:scorefile my_output_score/my_score.sc` should work.

**Comment by roccomoretti** [comment]
Marking this as "closed", as I hope your question has been answered. If that assessment is incorrect, feel free to re-open or open up a new discussion with more details.
